Crash on `tsc` subcommand when building with `go build`

```
PS D:\typescript-go> .\built\local\tsgo.exe tsc ./blah.ts
```

```
panic: vfs: path "bundled:///libs/lib.d.ts" is not absolute

goroutine 8 [running]:
github.com/microsoft/typescript-go/internal/vfs.rootLength({0x4000012f48, 0x18})
        D:/typescript-go/internal/vfs/vfs.go:74 +0x88
github.com/microsoft/typescript-go/internal/vfs.splitPath({0x4000012cc0?, 0xd?})
        D:/typescript-go/internal/vfs/vfs.go:81 +0x2c
github.com/microsoft/typescript-go/internal/vfs.(*common).rootAndPath(0x7ff6b5ccb618, {0x4000012cc0?, 0x7300000001?})
        D:/typescript-go/internal/vfs/vfs.go:88 +0x2c
github.com/microsoft/typescript-go/internal/vfs.(*common).ReadFile(0x18?, {0x4000012cc0?, 0x0?})
        D:/typescript-go/internal/vfs/vfs.go:157 +0x20
github.com/microsoft/typescript-go/internal/vfs.(*osFS).ReadFile(0x7ff6b5ccb618, {0x4000012cc0, 0x18})
        D:/typescript-go/internal/vfs/os.go:77 +0x70
github.com/microsoft/typescript-go/internal/compiler.(*compilerHost).GetSourceFile(0x4000012cc0?, {0x4000012cc0, 0x18}, 0x1)
        D:/typescript-go/internal/compiler/host.go:65 +0x34
github.com/microsoft/typescript-go/internal/compiler.(*fileLoader).parseSourceFile(0x4000000d80, {0x4000012cc0, 0x18})
        D:/typescript-go/internal/compiler/fileloader.go:218 +0xc4
github.com/microsoft/typescript-go/internal/compiler.(*fileLoader).startTasks.(*parseTask).start.func1()
        D:/typescript-go/internal/compiler/fileloader.go:178 +0x38
github.com/microsoft/typescript-go/internal/core.(*WorkGroup).Run.func1()
        D:/typescript-go/internal/core/workgroup.go:25 +0xf4
created by github.com/microsoft/typescript-go/internal/core.(*WorkGroup).Run in goroutine 1
        D:/typescript-go/internal/core/workgroup.go:19 +0x7c
```

--- a/cmd/tsgo/main.go
+++ b/cmd/tsgo/main.go
@@ -154,15 +154,14 @@ func main() {
 
 	currentDirectory = tspath.GetDirectoryPath(configFilePath)
 	// !!! is the working directory actually the config path?
-	host := ts.NewCompilerHost(compilerOptions, currentDirectory, fs)
+	host := ts.NewCompilerHost(compilerOptions, currentDirectory, fs, defaultLibraryPath)
 
 	parseStart := time.Now()
 	program := ts.NewProgram(ts.ProgramOptions{
-		ConfigFilePath:     configFilePath,
-		Options:            compilerOptions,
-		SingleThreaded:     opts.devel.singleThreaded,
-		Host:               host,
-		DefaultLibraryPath: defaultLibraryPath,
+		ConfigFilePath: configFilePath,
+		Options:        compilerOptions,
+		SingleThreaded: opts.devel.singleThreaded,
+		Host:           host,
 	})
 	parseTime := time.Since(parseStart)
 

--- a/cmd/tsgo/sys.go
+++ b/cmd/tsgo/sys.go
@@ -6,6 +6,7 @@ import (
 	"os"
 	"runtime"
 
+	"github.com/microsoft/typescript-go/internal/bundled"
 	"github.com/microsoft/typescript-go/internal/core"
 	"github.com/microsoft/typescript-go/internal/execute"
 	"github.com/microsoft/typescript-go/internal/tspath"
@@ -13,14 +14,19 @@ import (
 )
 
 type osSys struct {
-	writer  io.Writer
-	fs      vfs.FS
-	newLine string
-	cwd     string
+	writer             io.Writer
+	fs                 vfs.FS
+	defaultLibraryPath string
+	newLine            string
+	cwd                string
 }
 
 func (s *osSys) FS() vfs.FS {
 	return s.fs
+}
+
+func (s *osSys) DefaultLibraryPath() string {
+	return s.defaultLibraryPath
 }
 
 func (s *osSys) GetCurrentDirectory() string {
@@ -48,9 +54,10 @@ func newSystem() *osSys {
 	}
 
 	return &osSys{
-		cwd:     tspath.NormalizePath(cwd),
-		fs:      vfs.FromOS(),
-		writer:  os.Stdout,
-		newLine: core.IfElse(runtime.GOOS == "windows", "\r\n", "\n"),
+		cwd:                tspath.NormalizePath(cwd),
+		fs:                 bundled.WrapFS(vfs.FromOS()),
+		defaultLibraryPath: bundled.LibPath(),
+		writer:             os.Stdout,
+		newLine:            core.IfElse(runtime.GOOS == "windows", "\r\n", "\n"),
 	}
 }

--- a/internal/checker/checker_test.go
+++ b/internal/checker/checker_test.go
@@ -36,11 +36,10 @@ foo.bar;`
 	fs = bundled.WrapFS(fs)
 
 	cd := "/"
-	host := compiler.NewCompilerHost(nil, cd, fs)
+	host := compiler.NewCompilerHost(nil, cd, fs, bundled.LibPath())
 	opts := compiler.ProgramOptions{
-		Host:               host,
-		ConfigFilePath:     "/tsconfig.json",
-		DefaultLibraryPath: bundled.LibPath(),
+		Host:           host,
+		ConfigFilePath: "/tsconfig.json",
 	}
 	p := compiler.NewProgram(opts)
 	p.BindSourceFiles()
@@ -67,11 +66,10 @@ func TestCheckSrcCompiler(t *testing.T) {
 
 	rootPath := tspath.CombinePaths(tspath.NormalizeSlashes(repo.TypeScriptSubmodulePath), "src", "compiler")
 
-	host := compiler.NewCompilerHost(nil, rootPath, fs)
+	host := compiler.NewCompilerHost(nil, rootPath, fs, bundled.LibPath())
 	opts := compiler.ProgramOptions{
-		Host:               host,
-		ConfigFilePath:     tspath.CombinePaths(rootPath, "tsconfig.json"),
-		DefaultLibraryPath: bundled.LibPath(),
+		Host:           host,
+		ConfigFilePath: tspath.CombinePaths(rootPath, "tsconfig.json"),
 	}
 	p := compiler.NewProgram(opts)
 	p.CheckSourceFiles()

--- a/internal/compiler/fileloader.go
+++ b/internal/compiler/fileloader.go
@@ -46,7 +46,7 @@ func processAllProgramFiles(
 		compilerOptions:    compilerOptions,
 		resolver:           resolver,
 		tasksByFileName:    make(map[string]*parseTask),
-		defaultLibraryPath: programOptions.DefaultLibraryPath,
+		defaultLibraryPath: host.DefaultLibraryPath(),
 		comparePathsOptions: tspath.ComparePathsOptions{
 			UseCaseSensitiveFileNames: host.FS().UseCaseSensitiveFileNames(),
 			CurrentDirectory:          host.GetCurrentDirectory(),

--- a/internal/compiler/host.go
+++ b/internal/compiler/host.go
@@ -11,6 +11,7 @@ import (
 
 type CompilerHost interface {
 	FS() vfs.FS
+	DefaultLibraryPath() string
 	GetCurrentDirectory() string
 	NewLine() string
 	Trace(msg string)
@@ -25,21 +26,27 @@ type FileInfo struct {
 var _ CompilerHost = (*compilerHost)(nil)
 
 type compilerHost struct {
-	options          *core.CompilerOptions
-	currentDirectory string
-	fs               vfs.FS
+	options            *core.CompilerOptions
+	currentDirectory   string
+	fs                 vfs.FS
+	defaultLibraryPath string
 }
 
-func NewCompilerHost(options *core.CompilerOptions, currentDirectory string, fs vfs.FS) CompilerHost {
+func NewCompilerHost(options *core.CompilerOptions, currentDirectory string, fs vfs.FS, defaultLibraryPath string) CompilerHost {
 	h := &compilerHost{}
 	h.options = options
 	h.currentDirectory = currentDirectory
 	h.fs = fs
+	h.defaultLibraryPath = defaultLibraryPath
 	return h
 }
 
 func (h *compilerHost) FS() vfs.FS {
 	return h.fs
+}
+
+func (h *compilerHost) DefaultLibraryPath() string {
+	return h.defaultLibraryPath
 }
 
 func (h *compilerHost) SetOptions(options *core.CompilerOptions) {

--- a/internal/compiler/program.go
+++ b/internal/compiler/program.go
@@ -7,7 +7,6 @@ import (
 
 	"github.com/microsoft/typescript-go/internal/ast"
 	"github.com/microsoft/typescript-go/internal/binder"
-	"github.com/microsoft/typescript-go/internal/bundled"
 	"github.com/microsoft/typescript-go/internal/checker"
 	"github.com/microsoft/typescript-go/internal/compiler/diagnostics"
 	"github.com/microsoft/typescript-go/internal/compiler/module"
@@ -27,7 +26,6 @@ type ProgramOptions struct {
 	Options            *core.CompilerOptions
 	SingleThreaded     bool
 	ProjectReference   []core.ProjectReference
-	DefaultLibraryPath string
 	OptionsDiagnostics []*ast.Diagnostic
 }
 
@@ -46,7 +44,6 @@ type Program struct {
 	resolvedModules map[tspath.Path]module.ModeAwareCache[*module.ResolvedModule]
 
 	comparePathsOptions tspath.ComparePathsOptions
-	defaultLibraryPath  string
 
 	files       []*ast.SourceFile
 	filesByPath map[tspath.Path]*ast.SourceFile
@@ -87,11 +84,6 @@ func NewProgram(options ProgramOptions) *Program {
 	p.host = options.Host
 	if p.host == nil {
 		panic("host required")
-	}
-
-	p.defaultLibraryPath = options.DefaultLibraryPath
-	if p.defaultLibraryPath == "" {
-		panic("default library path required")
 	}
 
 	rootFiles := options.RootFiles
@@ -143,12 +135,12 @@ func NewProgram(options ProgramOptions) *Program {
 	if p.compilerOptions.NoLib != core.TSTrue {
 		if p.compilerOptions.Lib == nil {
 			name := tsoptions.GetDefaultLibFileName(p.compilerOptions)
-			libs = append(libs, tspath.CombinePaths(p.defaultLibraryPath, name))
+			libs = append(libs, tspath.CombinePaths(p.host.DefaultLibraryPath(), name))
 		} else {
 			for _, lib := range p.compilerOptions.Lib {
 				name, ok := tsoptions.GetLibFileName(lib)
 				if ok {
-					libs = append(libs, tspath.CombinePaths(p.defaultLibraryPath, name))
+					libs = append(libs, tspath.CombinePaths(p.host.DefaultLibraryPath(), name))
 				}
 				// !!! error on unknown name
 			}
@@ -171,7 +163,6 @@ func NewProgramFromParsedCommandLine(config *tsoptions.ParsedCommandLine, host C
 		Host:      host,
 		// todo: ProjectReferences
 		OptionsDiagnostics: config.GetConfigFileParsingDiagnostics(),
-		DefaultLibraryPath: bundled.LibPath(),
 	}
 	return NewProgram(programOptions)
 }

--- a/internal/compiler/program_test.go
+++ b/internal/compiler/program_test.go
@@ -224,11 +224,10 @@ func TestProgram(t *testing.T) {
 			opts := core.CompilerOptions{Target: testCase.target}
 
 			program := NewProgram(ProgramOptions{
-				RootFiles:          []string{"c:/dev/src/index.ts"},
-				Host:               NewCompilerHost(&opts, "c:/dev/src", fs),
-				Options:            &opts,
-				DefaultLibraryPath: bundled.LibPath(),
-				SingleThreaded:     false,
+				RootFiles:      []string{"c:/dev/src/index.ts"},
+				Host:           NewCompilerHost(&opts, "c:/dev/src", fs, bundled.LibPath()),
+				Options:        &opts,
+				SingleThreaded: false,
 			})
 
 			actualFiles := []string{}

--- a/internal/execute/system.go
+++ b/internal/execute/system.go
@@ -10,6 +10,7 @@ type System interface {
 	Writer() io.Writer
 	EndWrite() // needed for testing
 	FS() vfs.FS
+	DefaultLibraryPath() string
 	GetCurrentDirectory() string
 	NewLine() string // #241 eventually we want to use "\n"
 }

--- a/internal/execute/testsys_test.go
+++ b/internal/execute/testsys_test.go
@@ -27,26 +27,32 @@ func newTestSys(fileOrFolderList FileMap, cwd string, args ...string) *testSys {
 	}
 	fs := bundled.WrapFS(vfstest.FromMapFS(mapFS, true /*useCaseSensitiveFileNames*/))
 	return &testSys{
-		fs:           fs,
-		cwd:          cwd,
-		files:        fileList,
-		output:       []string{},
-		currentWrite: &strings.Builder{},
+		fs:                 fs,
+		defaultLibraryPath: bundled.LibPath(),
+		cwd:                cwd,
+		files:              fileList,
+		output:             []string{},
+		currentWrite:       &strings.Builder{},
 	}
 }
 
 type testSys struct {
 	// todo: original has write to output as a string[] because the separations are needed for baselining
-	output         []string
-	currentWrite   *strings.Builder
-	serializedDiff map[string]string
-	fs             vfs.FS
-	cwd            string
-	files          []string
+	output             []string
+	currentWrite       *strings.Builder
+	serializedDiff     map[string]string
+	fs                 vfs.FS
+	defaultLibraryPath string
+	cwd                string
+	files              []string
 }
 
 func (s *testSys) FS() vfs.FS {
 	return s.fs
+}
+
+func (s *testSys) DefaultLibraryPath() string {
+	return s.defaultLibraryPath
 }
 
 func (s *testSys) GetCurrentDirectory() string {

--- a/internal/execute/tsc.go
+++ b/internal/execute/tsc.go
@@ -168,7 +168,7 @@ func getParsedCommandLineOfConfigFile(configFileName string, options *core.Compi
 }
 
 func performCompilation(sys System, cb cbType, config *tsoptions.ParsedCommandLine, reportDiagnostic diagnosticReporter) ExitStatus {
-	host := compiler.NewCompilerHost(config.CompilerOptions(), sys.GetCurrentDirectory(), sys.FS())
+	host := compiler.NewCompilerHost(config.CompilerOptions(), sys.GetCurrentDirectory(), sys.FS(), sys.DefaultLibraryPath())
 	// todo: cache, statistics, tracing
 	program := compiler.NewProgramFromParsedCommandLine(config, host)
 	options := program.Options()

--- a/internal/ls/host.go
+++ b/internal/ls/host.go
@@ -9,6 +9,7 @@ import (
 
 type Host interface {
 	FS() vfs.FS
+	DefaultLibraryPath() string
 	GetCurrentDirectory() string
 	NewLine() string
 	Trace(msg string)

--- a/internal/ls/languageservice.go
+++ b/internal/ls/languageservice.go
@@ -24,6 +24,11 @@ func (l *LanguageService) FS() vfs.FS {
 	return l.host.FS()
 }
 
+// DefaultLibraryPath implements compiler.CompilerHost.
+func (l *LanguageService) DefaultLibraryPath() string {
+	return l.host.DefaultLibraryPath()
+}
+
 // GetCurrentDirectory implements compiler.CompilerHost.
 func (l *LanguageService) GetCurrentDirectory() string {
 	return l.host.GetCurrentDirectory()

--- a/internal/lsp/server.go
+++ b/internal/lsp/server.go
@@ -68,6 +68,11 @@ func (s *Server) FS() vfs.FS {
 	return s.fs
 }
 
+// DefaultLibraryPath implements project.ProjectServiceHost.
+func (s *Server) DefaultLibraryPath() string {
+	return s.defaultLibraryPath
+}
+
 // GetCurrentDirectory implements project.ProjectServiceHost.
 func (s *Server) GetCurrentDirectory() string {
 	return s.cwd

--- a/internal/project/host.go
+++ b/internal/project/host.go
@@ -4,6 +4,7 @@ import "github.com/microsoft/typescript-go/internal/vfs"
 
 type ProjectServiceHost interface {
 	FS() vfs.FS
+	DefaultLibraryPath() string
 	GetCurrentDirectory() string
 	NewLine() string
 	Trace(msg string)

--- a/internal/project/project.go
+++ b/internal/project/project.go
@@ -90,6 +90,11 @@ func (p *Project) FS() vfs.FS {
 	return p.projectService.host.FS()
 }
 
+// DefaultLibraryPath implements LanguageServiceHost.
+func (p *Project) DefaultLibraryPath() string {
+	return p.projectService.host.DefaultLibraryPath()
+}
+
 // GetCompilerOptions implements LanguageServiceHost.
 func (p *Project) GetCompilerOptions() *core.CompilerOptions {
 	return p.compilerOptions
@@ -136,10 +141,9 @@ func (p *Project) GetProgram() *compiler.Program {
 	compilerOptions := p.GetCompilerOptions()
 
 	p.program = compiler.NewProgram(compiler.ProgramOptions{
-		RootFiles:          rootFileNames,
-		Host:               p,
-		Options:            compilerOptions,
-		DefaultLibraryPath: p.GetDefaultLibraryPath(),
+		RootFiles: rootFileNames,
+		Host:      p,
+		Options:   compilerOptions,
 	})
 
 	p.program.BindSourceFiles()

--- a/internal/testutil/harnessutil/harnessutil.go
+++ b/internal/testutil/harnessutil/harnessutil.go
@@ -158,7 +158,7 @@ func CompileFiles(
 	fs := vfstest.FromMapFS(testfs, harnessOptions.UseCaseSensitiveFileNames)
 	fs = bundled.WrapFS(fs)
 
-	host := createCompilerHost(fs, &compilerOptions, currentDirectory)
+	host := createCompilerHost(fs, bundled.LibPath(), &compilerOptions, currentDirectory)
 	result := compileFilesWithHost(host, programFileNames, &compilerOptions, &harnessOptions)
 
 	return result
@@ -350,8 +350,8 @@ func getOptionValue(t *testing.T, option *tsoptions.CommandLineOption, value str
 	return nil
 }
 
-func createCompilerHost(fs vfs.FS, options *core.CompilerOptions, currentDirectory string) compiler.CompilerHost {
-	return compiler.NewCompilerHost(options, currentDirectory, fs)
+func createCompilerHost(fs vfs.FS, defaultLibraryPath string, options *core.CompilerOptions, currentDirectory string) compiler.CompilerHost {
+	return compiler.NewCompilerHost(options, currentDirectory, fs, defaultLibraryPath)
 }
 
 func compileFilesWithHost(
@@ -446,10 +446,9 @@ func newCompilationResult(
 // !!! Temporary while we don't have the real `createProgram`
 func createProgram(host compiler.CompilerHost, options *core.CompilerOptions, rootFiles []string) *compiler.Program {
 	programOptions := compiler.ProgramOptions{
-		RootFiles:          rootFiles,
-		Host:               host,
-		Options:            options,
-		DefaultLibraryPath: bundled.LibPath(),
+		RootFiles: rootFiles,
+		Host:      host,
+		Options:   options,
 	}
 	program := compiler.NewProgram(programOptions)
 	return program

